crates/resolver/src/async_resolver.rs

```diff
@@ -58,15 +59,15 @@ type ClientCache<S> = CachingClient<LookupEither<Connection, StandardConnection,
 /// linked to it. When all of its [`AsyncResolver`]s have been dropped, the
 /// background future will finish.
 #[derive(Clone)]
-pub struct AsyncResolver<S: SpawnBg> {
+pub struct AsyncResolver<C: DnsHandle, P: ConnectionProvider<Conn = C>> {
```

Comment:
Ok, @chunyingw , I pulled ConnectionProvider up and dropped SpawnBg. What do you think?